pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -426,6 +464,9 @@ type BareMetalHostStatus struct {
 	// The hardware discovered to exist on the host.
 	HardwareDetails *HardwareDetails `json:"hardware,omitempty"`
 
+	// The RootDevicehints set by the user
+	RootDeviceHints RootDeviceHints `json: "rootDeviceHints,omitempty"`
```

Comment:
I think the point of this was to record what was used on the last provisioning operation. If so, maybe it should go into the Provisioning struct?